lookout/style/typos/analyzer.py

```diff
@@ -86,12 +93,15 @@ def analyze(self, ptr_from: ReferencePointer, ptr_to: ReferencePointer,
         :return: List of found review suggestions. Refer to \
                  lookout/core/server/sdk/service_analyzer.proto.
         """
-        return [generate_comment(
-            filename=typo_fix.head_file.path,
-            line=typo_fix.line_number,
-            text=self.render_comment_text(typo_fix),
-            confidence=self._get_comment_confidence(typo_fix.candidates))
-            for typo_fix in self.generate_typos_fixes(list(changes))]
+        comments = []
+        for typo_fix in self.generate_typos_fixes(list(changes)):
+            text, confidences = self.render_comment_text(typo_fix)
```

Comment:
I believe that now you do not need to return confidences from `render_comment_text` because you already have them in `typo_fix`. Let's return text only. 

In this case, the old code works with a small change:
```python
return [generate_comment(
            filename=typo_fix.head_file.path,
            line=typo_fix.line_number,
            text=self.render_comment_text(typo_fix),
            confidence=self._get_comment_confidence(typo_fix. identifiers))
            for typo_fix in self.generate_typos_fixes(list(changes))]
```
Up to you to take it back  or use your map-reduce formula :)